buildSrc/src/main/kotlin/com/toasttab/protokt/shared/ProtoktExtension.kt

```diff
@@ -16,28 +16,30 @@
 package com.toasttab.protokt.shared
 
 import java.util.jar.JarInputStream
+import kotlin.reflect.KClass
 
 const val DEFAULT_PROTOBUF_VERSION = "3.12.1"
 const val MANIFEST_VERSION_PROPERTY = "Implementation-Version"
 
 open class ProtoktExtension {
     var protocVersion = DEFAULT_PROTOBUF_VERSION
-    var toolsVersion: String? = null
     var publishProto = false
     var respectJavaPackage = true
 
     val version by lazy {
-        toolsVersion
-            ?: ProtoktExtension::class.java
-                .protectionDomain
-                .codeSource
-                .location
-                .openStream()
-                .use {
-                    JarInputStream(it)
-                        .manifest
-                        .mainAttributes
-                        .getValue(MANIFEST_VERSION_PROPERTY)
-                }
+        getProtoktVersion(ProtoktExtension::class)
     }
 }
+
+fun getProtoktVersion(klass: KClass<*>): String =
+    klass.java
```

Comment:
this code path is only used by the gradle plugin, not the local protokt build

so it can be moved into a tiny shared module instead of packaging buildSrc classes into codegen

Reply:
We still need access to the constants for options. That happened to work without adding to the JAR task, so those can stay put?

Reply:
also you can forego manifest generation and parsing and add something like https://github.com/crotwell/version-class to the codegen and plugin projects

Reply:
That plugin didn't work OOB and my best finagling with SourceSets couldn't get it working, so I'm going to defer that for now. I've extracted to a small module.

Reply:
... which unfortunately, must be published.